jupyter-extension/Dockerfile

```diff
@@ -31,4 +31,9 @@ USER $NB_UID
 COPY dist dist
 WORKDIR /home/jovyan
 RUN pip install `find /app/dist/ -name \*.whl` nbgitpuller
-RUN pip install determined==0.23.3
\ No newline at end of file
+RUN pip install determined==0.23.3
+RUN pip install jupyterlab_pachyderm
+RUN pip install --upgrade pip && \
```

Comment:
The pycurl package should have it's dependency pinned. Additionally, if the pycurl package is going to be used directly by the jupyterlab extension it should be included in the `setup.cfg` file instead of this Dockerfile.

Reply:
ok

Reply:
We need --compile option and also the export 
i.e. RUN export PYCURL_SSL_LIBRARY=openssl && \
    pip install --compile pycurl
I am not sure how to address this in setup.cfg, still looking.

Reply:
I tried something as follows and it does not work,

```
RUN export PYCURL_SSL_LIBRARY=openssl && \
    pip install --compile pycurl==7.45.2

```
Let me know if you have suggestions to do it some other way.

Reply:
Why is the compile flag needed? I would naïvely assume that if pip cannot install the package from its wheel then it would default to compiling.

If this compiling step is only needed due to this being packaged into a Docker image, you should be able to add `pycurl` as a dependency within the `setup.cfg` and then have this Dockerfile command precede the command which installs the jupyterlab extension wheel (line 33). However, I think we should be very careful about making this product even more difficult to install unless there are no other options.

Reply:
I was getting some errors if I did not use the flags as mentioned. The pycurl document has some discussion about it.
http://pycurl.io/docs/latest/install.html#pip-and-cached-pycurl-package
There is also a stackoverflow discussion about it.

Reply:
I moved it to ci-requirements.txt and pinned it there, works withou --compile flag now.